github_issue_summarization/distributed/train.py

```diff
@@ -14,164 +15,208 @@
 
-# Word embeding for encoder (ex: Issue Body)
-x = tf.keras.layers.Embedding(
-      num_encoder_tokens, latent_dim, name='Body-Word-Embedding', mask_zero=False)(encoder_inputs)
-x = tf.keras.layers.BatchNormalization(name='Encoder-Batchnorm-1')(x)
+  # Intermediate GRU layer (optional)
+  #x = GRU(latent_dim, name='Encoder-Intermediate-GRU', return_sequences=True)(x)
+  #x = BatchNormalization(name='Encoder-Batchnorm-2')(x)
 
-# Intermediate GRU layer (optional)
-#x = GRU(latent_dim, name='Encoder-Intermediate-GRU', return_sequences=True)(x)
-#x = BatchNormalization(name='Encoder-Batchnorm-2')(x)
+  # We do not need the `encoder_output` just the hidden state.
+  _, state_h = tf.keras.layers.GRU(latent_dim, return_state=True, name='Encoder-Last-GRU')(x)
 
-# We do not need the `encoder_output` just the hidden state.
-_, state_h = tf.keras.layers.GRU(latent_dim, return_state=True, name='Encoder-Last-GRU')(x)
+  # Encapsulate the encoder as a separate entity so we can just
+  #  encode without decoding if we want to.
 
-# Encapsulate the encoder as a separate entity so we can just
-#  encode without decoding if we want to.
-encoder_model = tf.keras.Model(inputs=encoder_inputs, outputs=state_h, name='Encoder-Model')
+  # TODO(jlewi): I commented out the following two lines
+  # encoder_model = tf.keras.Model(inputs=encoder_inputs, outputs=state_h, name='Encoder-Model')
+  # seq2seq_encoder_out = encoder_model(encoder_inputs)
 
-seq2seq_encoder_out = encoder_model(encoder_inputs)
+  ########################
+  #### Decoder Model ####
+  decoder_inputs = tf.keras.layers.Input(shape=(None,), name='Decoder-Input')  # for teacher forcing
 
-########################
-#### Decoder Model ####
-decoder_inputs = tf.keras.layers.Input(shape=(None,), name='Decoder-Input')  # for teacher forcing
+  # Word Embedding For Decoder (ex: Issue Titles)
+  dec_emb = tf.keras.layers.Embedding(
+              num_decoder_tokens,
+              latent_dim, name='Decoder-Word-Embedding',
+              mask_zero=False)(decoder_inputs)
+  dec_bn = tf.keras.layers.BatchNormalization(name='Decoder-Batchnorm-1')(dec_emb)
 
-# Word Embedding For Decoder (ex: Issue Titles)
-dec_emb = tf.keras.layers.Embedding(
-            num_decoder_tokens,
-            latent_dim, name='Decoder-Word-Embedding',
-            mask_zero=False)(decoder_inputs)
-dec_bn = tf.keras.layers.BatchNormalization(name='Decoder-Batchnorm-1')(dec_emb)
+  # Set up the decoder, using `decoder_state_input` as _state.
+  decoder_gru = tf.keras.layers.GRU(
+                  latent_dim, return_state=True, return_sequences=True, name='Decoder-GRU')
 
-# Set up the decoder, using `decoder_state_input` as _state.
-decoder_gru = tf.keras.layers.GRU(
-                latent_dim, return_state=True, return_sequences=True, name='Decoder-GRU')
+  # FIXME: seems to be running into this https://github.com/keras-team/keras/issues/9761
+  decoder_gru_output, _ = decoder_gru(dec_bn)  # , initial_state=seq2seq_encoder_out)
```

Comment:
@jlewi this appears to be a bug.  The decoder should be initialized with the last hidden state of the encoder.  Here it is not being initialized with anything.  I understand there may have been some breaking changes to the keras api per the #FIXME comments so I would suggest holding this PR for now until a work around is resolved.  This relates to your question about seq2seq encoder out being used.

Reply:
Got it thanks.